pittapi/course_helper.py

```diff
@@ -9,33 +9,34 @@
     # id -> unique course ID, not to be confused with course code (for instance, CS 0007 has code 105611)
     # career -> for example, UGRD (undergraduate)
 
+valid_terms = re.compile("2\d\d[147]")
 
-def _validate_term(term: str) -> str:
+
+def _validate_term(term: Union[str, int]) -> str:
     """Validates that the term entered follows the pattern that Pitt does for term codes."""
-    valid_terms = re.compile("2\d\d[147]")
     if valid_terms.match(str(term)):
         return str(term)
-    raise ValueError("Term entered isn't a valid Pitt term.")
+    raise ValueError("Term entered isn't a valid Pitt term, must match regex 2\d\d[147]")
```

Comment:
This error message has the potential to get out-of-sync with the regex used for validation if the regex ever needs to change. While it would typically be advisable to describe the pattern in words, here it would be overly verbose to fully describe the century code (yes, that's what the `2` is!), year code, and term code structure. So it may be better here to store the expected regex string in a variable and refer to its contents directly in the error message to avoid duplication.

Reply:
Addressed in recent commit. Thanks for the feedback!!